src/analysis/retail/priest/discipline/modules/guide/EvangelismAnalysis.tsx

```diff
@@ -0,0 +1,125 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_PRIEST } from 'common/TALENTS';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import CASTS_THAT_ARENT_CASTS from 'parser/core/CASTS_THAT_ARENT_CASTS';
+import Events, { CastEvent } from 'parser/core/Events';
+import { Options } from 'parser/core/Module';
+import EventHistory from 'parser/shared/modules/EventHistory';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import GlobalCooldown from '../core/GlobalCooldown';
+import Atonement from '../spells/Atonement';
+import Evangelism from '../spells/Evangelism';
+import Haste from 'parser/shared/modules/Haste';
+import { SpellLink } from 'interface';
+import { CooldownExpandable } from 'analysis/retail/druid/restoration/Guide';
+
+const ALLOWED_PRE_EVANG = [
+  TALENTS_PRIEST.POWER_WORD_RADIANCE_TALENT.id,
+  SPELLS.POWER_WORD_SHIELD.id,
+  TALENTS_PRIEST.RENEW_TALENT.id,
+  SPELLS.FLASH_HEAL.id,
+  TALENTS_PRIEST.RAPTURE_TALENT.id,
+  TALENTS_PRIEST.SHADOWFIEND_TALENT.id,
+  TALENTS_PRIEST.EVANGELISM_TALENT.id,
+  SPELLS.SHADOW_WORD_PAIN.id,
+  TALENTS_PRIEST.PURGE_THE_WICKED_TALENT.id,
+];
+
+const PEMITTED_RAMP_STARTERS = [
+  SPELLS.SHADOW_WORD_PAIN.id,
+  TALENTS_PRIEST.PURGE_THE_WICKED_TALENT.id,
+  TALENTS_PRIEST.RENEW_TALENT.id,
+  SPELLS.FLASH_HEAL.id,
+  TALENTS_PRIEST.RAPTURE_TALENT.id,
+  SPELLS.POWER_WORD_SHIELD.id,
+];
+
+interface Ramp {
+  timestamp: number;
+  badCastIndexes?: number[];
+}
```

Comment:
You need `rampHistory` here too right? In order for the indices to mean anything

Reply:
Yup, great spot! 

I've added it now :)